src/qibolab/designs/channels.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from qibolab.instruments.abstract import LocalOscillator
+
+
+@dataclass
+class Channel:
+    """Representation of physical wire connection (channel)."""
+
+    name: str
+    """Name of the channel from the lab schematics."""
+
+    qubit: Optional["Qubit"] = field(default=None, repr=False)
+    """Qubit connected to this channel.
+    Used to read the sweetspot and filters for flux channels only. ``None`` for non-flux channels.
+    """
+    ports: List[tuple] = field(default_factory=list)
+    """List of tuples (controller, port) connected to this channel."""
+    local_oscillator: Optional[LocalOscillator] = None
+    """Instrument object controlling the local oscillator connected to this channel.
+    Not applicable for setups that do not use local oscillators because the controller
+    can send sufficiently high frequencies
+    """
+    _offset: Optional[float] = None
+    _filter: Optional[dict] = None
+
+    @property
+    def offset(self):
+        """Bias offset for flux channels."""
+        if self._offset is None:
+            # operate qubits at their sweetspot unless otherwise stated
+            return self.qubit.sweetspot
+        return self._offset
+
+    @offset.setter
+    def offset(self, offset):
+        self._offset = offset
+
+    @property
+    def filter(self):
+        """Filters for sending flux pulses through a flux channel."""
+        if self._filter is None:
+            return self.qubit.filter
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        self._filter = filter
```

Comment:
Same as above.

Reply:
Added type check for dict, but same problem as above. In fact `filter` may not be useful for platforms other than QM. My plan was to see if that's the case when we do other drivers and if it is really a QM thing I would remove it from the `Channel` altogether.